src/services/highcharts-formatter/highcharts-formatter.svc.coffee

```diff
@@ -1,103 +1,101 @@
 angular
 .module('impac.services.highcharts-formatter', [])
-.service('HighChartsFormatter', ($filter) ->
-  _self = @
+.factory('HighchartsFactory', ($filter)->
 
-  todayMarker = (dates, showToday) ->
-    return {} unless showToday
-    projection_date = _.find(dates, (date)-> moment(date) >= moment().startOf('day'))
-    todayIndex = _.indexOf(dates, projection_date)
-
-    xAxis:
-      plotLines: [{
-        color: 'rgba(0, 85, 255, 0.2)'
-        value: todayIndex
-        width: 1
-        label:
-          text: null
-          verticalAlign: 'top'
-          textAlign: 'center'
-          rotation: 0
-          y: -5
-      }]
+  templates =
+    line: Object.freeze
+      get: (options = {})->
+        chart:
+          type: 'line'
+          zoomType: 'x'
+          spacingTop: 20
+        title: null
+        credits:
+          enabled: false
+        legend:
+          enabled: _.get(options, 'showLegend', true)
+          layout: 'vertical'
+          align: 'left'
+          verticalAlign: 'middle'
+        xAxis:
+          startOnTick: false
+          minPadding: 0
+          tickInterval: 1
+          min: 0
+        yAxis:
+          title: null
+          startOnTick: true
+          minPadding: 0
+        series: _.get(options, 'data.series', [])
 
-  thresholdsMarkers = (thresholds) ->
-    return {} if _.isEmpty(thresholds)
-    lines = []
-    for threshold in thresholds
-      lines.push({
-        color: 'rgba(255, 0, 0, 0.5)'
-        value: threshold
-        width: 2
-        zIndex: 5
-      })
+  class Chart
+    constructor: (@id, @options = {})->
+      @_template = templates[@options.chartType]
+      return
 
-    yAxis:
-      plotLines: lines
+    render: (options)->
+      angular.merge(@options, options)
+      chartConfig = angular.merge({},
+        @template(@options)
```

Comment:
Not a big fan of passing all the options as parameters when you just need some of them, but I guess this won't make a big difference